### PR TITLE
DOC: Replace .format by f-strings in examples

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

Just some minor modernization.

## PR Checklist

**Documentation and Tests**
- [n/a] Has pytest style unit tests (and `pytest` passes)
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [n/a] New plotting related features are documented with examples.

**Release Notes**
- [n/a] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [n/a] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [n/a] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

**Comment by QuLogic** [comment]
Yay, looks like that fixes the CodeQL warning thing, too.

**Review by dstansby — APPROVED** [review]

File: examples/text_labels_and_annotations/font_table.py [hunk 1]
```diff
@@ -75,8 +75,8 @@ def draw_font_table(path):
     # we have indeed selected a Unicode charmap.
     codes = font.get_charmap().items()
 
-    labelc = ["{:X}".format(i) for i in range(16)]
-    labelr = ["{:02X}".format(16 * i) for i in range(16)]
+    labelc = [f"{i:X}" for i in range(16)]
+    labelr = [f"{i:02X}" for i in range(0, 16*16, 16)]
```
Comment (greglucas) [inline examples/text_labels_and_annotations/font_table.py:79]:
```suggestion
    labelr = [f"{i * 16:02X}" for i in range(16)]
```
Seems more confusing to do the range with offset?

**Review by greglucas — APPROVED** [review]

File: examples/text_labels_and_annotations/font_table.py (hunk 1 above)
Comment (QuLogic) [reply examples/text_labels_and_annotations/font_table.py:79]: Weirdly, I thought the other way was more confusing.

Comment (greglucas) [reply examples/text_labels_and_annotations/font_table.py:79]: :) sounds good, fine either way with me. This is definitely an improvement as is.

**Pull request merged**